sessiontxn/staleread/processor.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package staleread
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessiontxn"
+	"github.com/pingcap/tidb/table/temptable"
+)
+
+// StalenessTSEvaluator is a function to get staleness ts
+type StalenessTSEvaluator func(sctx sessionctx.Context) (uint64, error)
+
+// Processor is an interface used to process stale read
+type Processor interface {
```

Comment:
Consider `var _ Proccessor = &staleReadProcessor{}` to enforce the implementation?

Reply:
fine. Good idea